group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -8641,6 +8665,130 @@ public void testListGroups() {
+            new DescribeGroupsResponseData.DescribedGroup()
+                .setGroupId("group-id")
+                .setGroupState(STABLE.toString())
+                .setProtocolType(responseData.protocolType())
+                .setProtocolData(responseData.protocolName())
+                .setMembers(responseData.members().stream().map(member ->
+                    new DescribeGroupsResponseData.DescribedGroupMember()
+                        .setMemberId(member.memberId())
+                        .setGroupInstanceId(member.groupInstanceId())
+                        .setClientId("client")
+                        .setClientHost("localhost/127.0.0.1")
+                        .setMemberMetadata(member.metadata())
+                ).collect(Collectors.toList()))
+        );
+
+        List<DescribeGroupsResponseData.DescribedGroup> describedGroups =
+            context.describeGroups(Collections.singletonList("group-id"));
+
+        assertEquals(expectedDescribedGroups, describedGroups);
+    }
+
+    @Test
+    public void testDescribeGroupStableForStaticMember() throws Exception {
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .build();
+        context.staticMembersJoinAndRebalance(
+            "group-id",
+            "leader-instance-id",
+            "follower-instance-id"
+        );
+        context.testDescribeGroupsUncommittedGroup("group-id");
+
+        context.createSnapshot();
+        context.commit();
+
+        GenericGroup group = context.groupMetadataManager.getOrMaybeCreateGenericGroup("group-id", false);
+        List<DescribeGroupsResponseData.DescribedGroup> expectedDescribedGroups = Collections.singletonList(
+            new DescribeGroupsResponseData.DescribedGroup()
+                .setGroupId("group-id")
+                .setGroupState(STABLE.toString())
+                .setProtocolType(group.protocolType().get())
+                .setProtocolData(group.protocolName().get())
+                .setMembers(group.allMembers().stream().map(member ->
+                    new DescribeGroupsResponseData.DescribedGroupMember()
+                        .setMemberId(member.memberId())
+                        .setGroupInstanceId(member.groupInstanceId().get())
+                        .setClientId(member.clientId())
+                        .setClientHost(member.clientHost())
+                        .setMemberMetadata(member.metadata(group.protocolName().get()))
+                        .setMemberAssignment(member.assignment())
+                ).collect(Collectors.toList()))
+        );
+
+        List<DescribeGroupsResponseData.DescribedGroup> describedGroups =
+            context.describeGroups(Collections.singletonList("group-id"));
+
+        assertEquals(expectedDescribedGroups, describedGroups);
+    }
+
+    @Test
+    public void testDescribeGroupRebalancing() throws Exception {
```

Comment:
In this case, you can probably replay and then trigger a rebalance to get the state that you want.